### PR TITLE
fix: Fix code and update test sink/statistics-files-001

### PR DESCRIPTION
**Pull request by michaljurecko**
Jira: https://keboola.atlassian.net/browse/PSGO-596

All tests will pass in the final [PR](https://github.com/keboola/keboola-as-code/pull/1770).

[Large set of fixes and E2e tests](https://github.com/keboola/keboola-as-code/pull/1759) is divided into smaller PRs to do a review.

--------------

**Changes:**
- Updated Sink statistics code and tests.

-----------

Commend per commits. Please, explore [the final version](https://github.com/keboola/keboola-as-code/pull/1759) in the IDE.

-------------

### CODE REVIEW

File: internal/pkg/service/stream/api/service/sink.go [hunk 1]
```diff
@@ -239,6 +241,14 @@ func (s *service) SinkStatisticsFiles(ctx context.Context, d dependencies.SinkRe
...
 			ImportedAt:  timeToString(value.ImportedAt),
+			Statistics: &stream.SinkFileStatistics{
+				Levels: &stream.Levels{
+					Local:   &stream.Level{},
+					Staging: &stream.Level{},
+					Target:  &stream.Level{},
+				},
+				Total: &stream.Level{},
+			},
 		}
```
Comment (michaljurecko) [inline internal/pkg/service/stream/api/service/sink.go:252]: Initialization moved from the code bllow.

File: internal/pkg/service/stream/api/service/sink.go [hunk 2]
```diff
@@ -254,34 +264,26 @@ func (s *service) SinkStatisticsFiles(ctx context.Context, d dependencies.SinkRe
 		return nil, err
 	}
 
+	// Sort keys lexicographically
 	keys := maps.Keys(filesMap)
+	slices.SortStableFunc(keys, func(a, b model.FileID) int {
+		return strings.Compare(a.String(), b.String())
+	})
+
 	if len(keys) > 0 {
```
Comment (michaljurecko) [inline internal/pkg/service/stream/api/service/sink.go:273]: Go maps don't store order :exclamation: , see comments of `maps.*` functions.

File: internal/pkg/service/stream/api/service/sink.go [hunk 3]
```diff
@@ -254,34 +264,26 @@ func (s *service) SinkStatisticsFiles(ctx context.Context, d dependencies.SinkRe
...
+			assignStatistics(filesMap[k].Statistics.Levels.Staging, aggregated.Staging)
+			assignStatistics(filesMap[k].Statistics.Levels.Target, aggregated.Target)
 		}
 
 		for _, file := range filesMap {
-			if file.Statistics.Total.RecordsCount == 0 {
-				file.Statistics.Total = nil
-			}
 			if file.Statistics.Levels.Local.RecordsCount == 0 {
 				file.Statistics.Levels.Local = nil
```
Comment (michaljurecko) [inline internal/pkg/service/stream/api/service/sink.go:288]: Always include total count.

File: test/stream/api/sink/statistics-files-001-ok/001-create-source/expected-response.json [hunk 4]
```diff
@@ -0,0 +1,8 @@
+{
+  "taskId": "api.create.source/%%TEST_DEFAULT_BRANCH_ID%%/my-source/%s",
+  "type": "api.create.source",
+  "url": "https://stream.keboola.local/v1/tasks/api.create.source/%%TEST_DEFAULT_BRANCH_ID%%/my-source/%s",
```
Comment (michaljurecko) [inline test/stream/api/sink/statistics-files-001-ok/001-create-source/expected-response.json:4]: Resources are now created by API calls.

File: test/stream/api/sink/statistics-files-001-ok/initial-etcd-kvs.txt [hunk 5]
```diff
@@ -1,107 +1,3 @@
-<<<<<
-definition/branch/active/%%TEST_KBC_PROJECT_ID%%/%%TEST_DEFAULT_BRANCH_ID%%
------
-{
-  "projectId": %%TEST_KBC_PROJECT_ID%%,
-  "branchId": %%TEST_DEFAULT_BRANCH_ID%%,
-  "isDefault": true
```
Comment (michaljurecko) [inline test/stream/api/sink/statistics-files-001-ok/initial-etcd-kvs.txt:7]: So following part of the etcd initial state is not needed.

File: test/stream/api/sink/statistics-files-002-not-found/001-statistics-files/expected-response.json [hunk 6]
```diff
@@ -1,5 +1,5 @@
 {
   "statusCode": 404,
-  "error": "stream.api.sinkNotFound",
-  "message": "Sink \"my-sink\" not found in the source."
+  "error": "stream.api.sourceNotFound",
+  "message": "Source \"my-source\" not found in the branch."
```
Comment (michaljurecko) [inline test/stream/api/sink/statistics-files-002-not-found/001-statistics-files/expected-response.json:4]: API checks also if the source exists, the message is now better.

**Review by Matovidlo — APPROVED** [review]
LGTM

**Review by hosekpeter — APPROVED** [review]

**Pull request merged**